ext/com/concurrent_ruby/ext/JavaAtomicBooleanLibrary.java

```diff
@@ -0,0 +1,96 @@
+package com.concurrent_ruby.ext;
+
+import java.io.IOException;
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyModule;
+import org.jruby.RubyObject;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.jruby.runtime.load.Library;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.jruby.RubyBoolean;
+import org.jruby.RubyNil;
+import org.jruby.runtime.ThreadContext;
+
+public class JavaAtomicBooleanLibrary implements Library {
+
+    public void load(Ruby runtime, boolean wrap) throws IOException {
+        RubyModule concurrentMod = runtime.defineModule("Concurrent");
+        RubyClass atomicCls = concurrentMod.defineClassUnder("JavaAtomicBoolean", runtime.getObject(), JRUBYREFERENCE_ALLOCATOR);
+        atomicCls.defineAnnotatedMethods(JavaAtomicBoolean.class);
+    }
+
+    private static final ObjectAllocator JRUBYREFERENCE_ALLOCATOR = new ObjectAllocator() {
+        public IRubyObject allocate(Ruby runtime, RubyClass klazz) {
+            return new JavaAtomicBoolean(runtime, klazz);
+        }
+    };
+
+    @JRubyClass(name = "JavaAtomicBoolean", parent = "Object")
+    public static class JavaAtomicBoolean extends RubyObject {
+
+        private AtomicBoolean atomicBoolean;
+        private ThreadContext context;
```

Comment:
Sorry, I've missed this on first time. I believe this should not be stored in the object's field. The `ThreadContext` can change on each method call, it should be always first argument of the `@JRubyMethod`s.